eta/core/learning.py

```diff
@@ -623,6 +717,24 @@ def detect(self, imgs):
         '''
         raise NotImplementedError("subclasses must implement detect()")
 
+    def process(self, imgs):
+        '''Generates labels for the given tensor of images.
+
+        Args:
+            imgs: a list (or d x ny x nx x 3 tensor) of images to process
+
+        Returns:
+            an `eta.core.video.VideoLabels` instance containing the labels
+                generated for the given tensor of images
+        '''
+        labels = etav.VideoLabels()
+        attrs = self.detect(imgs)
```

Comment:
This returns a `DetectedObjectContainer`, not an `AttributeContainer`.

As you pointed out, its not quite clear how to embed these results in a `VideoLabels` because "video frames" models output intermediate results that are used in the context of a larger algorithm. We should chat so I can understand more about what your use case is for this.

For example, the idea is that a `VideoFramesObjectDetector` would be responsible for including `DetectedObjects` with frame numbers to indicate which frame of the input tensor the detection corresponds to. So, this method should partition the `DetectedObjects` into their respective frames.

With `VideoFramesClassifier`, your strategy of inserting the attributes into each frame _is_ the intended interpretation of the output of that model.

Reply:
Late response, apologies. I can chat whenever. And thanks for you thoughts!

I think I may just close this PR. The intention was to have a common interface for Models, but it is not turning out to be that useful.